### PR TITLE
[dev] upgrade rust-toolchain to 1.64, mold to 1.5.1, remove @action-rs/toolchain

### PR DESCRIPTION
**Pull request by jiangying000**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: should also fix https://github.com/starcoinorg/starcoin/issues/3508
## What is the new behavior?

-
-
-

## Other information

remove @action-rs/toolchain since it's unmaintained

https://github.com/actions-rs/toolchain/issues/216

### CODE REVIEW

File: .github/workflows/release_asset.yml [hunk 1]
```diff
@@ -20,8 +20,8 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-      - name: install rust toolchain
-        uses: actions-rs/toolchain@v1
+      # - name: install rust toolchain
+      #   uses: actions-rs/toolchain@v1
```
Comment (jiangying000) [inline .github/workflows/release_asset.yml:24]: 这里不确定

但是在benchmark_pr, codecov, build_and_test 里，把 actions-rs/toolchain@v1 移除都没用问题

Comment (jolestar) [reply .github/workflows/release_asset.yml:24]: 能移除 toolchain 的前提是 builder image 里包含了 rust 吧。release 这个用的是 github 的服务，应该只有一个空的 image，应该是要留着。

File: rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
```
Comment (jolestar) [inline rust-toolchain.toml:1]: rust-toolchain 文件名应该是约定俗成很多工具都用的吧？

Comment (jiangying000) [reply rust-toolchain.toml:1]: 按照文档原来的格式属于 lagecy 了，https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file:~:text=For%20backwards%20compatibility%2C,rust%2Dtoolchain.toml%20files.

文件名上加上 .toml 会有语法高亮提示，而且看起来和其他文件比较统一，感觉更好些，顺便可以筛选出很久没有更新的工具

已经有不少项目用了 rust-toolchain.toml（应该不会有坑），比如：
https://github.com/rustdesk/rustdesk/blob/master/rust-toolchain.toml
https://github.com/apollographql/router/blob/dev/rust-toolchain.toml
https://github.com/ChainSafe/forest/blob/main/rust-toolchain.toml
https://github.com/volta-cli/volta/blob/main/rust-toolchain.toml
https://github.com/rust-osdev/uefi-rs/blob/main/rust-toolchain.toml
https://github.com/casper-network/casper-node/blob/dev/rust-toolchain.toml

File: .github/workflows/release_asset.yml (hunk 1 above)
Comment (jiangying000) [reply .github/workflows/release_asset.yml:24]: 我找到了一个 actions/runner-images 仓库，应该是 github runner 的仓库，我把矩阵里面的系统都查了一遍，应该是是默认安装了的：
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu1804-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#rust-tools

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -15,18 +15,13 @@ jobs:
...
-          profile: minimal
-          components: rustfmt,clippy
       - name: run fmt check
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: -- --check
       - name: setup environment
-        run: bash ./scripts/dev_setup.sh  -b -t -y -m
+        run: bash ./scripts/dev_setup.sh  -b -t -y
```
Comment (jiangying000) [inline .github/workflows/build_test.yml:24]: 这里因为 mold 没有启用，所以不用安装

Comment (jolestar) [reply .github/workflows/build_test.yml:24]: > 这里因为 mold 没有启用，所以不用安装

记得 mold 是在 linux 下自动启用的？ @LemonHX

Comment (jiangying000) [reply .github/workflows/build_test.yml:24]: https://github.com/starcoinorg/starcoin/blob/1bd6874a871e73ef4992d2014cf26e23c412d406/.cargo/config#L17-L22

相关配置注释掉了

Comment (jolestar) [reply .github/workflows/build_test.yml:24]: 脚本里会改

https://github.com/starcoinorg/starcoin/blob/1bd6874a871e73ef4992d2014cf26e23c412d406/scripts/dev_setup.sh#L538-L559

Comment (jiangying000) [reply .github/workflows/build_test.yml:24]: 收到

**Review by jolestar — APPROVED** [review]

**Pull request merged**